### ISSUE TITLE
Dev: Investigate technical requirements for leaving comments on a project.

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#1840]
Leaving comments is a common (but optional) aspect of working on a shared document. What are the technical pieces needed to make this work assuming that:

* Collaborators can leave text-based comments that are attached to a file and line number (for example)

**Comment by Pomax** [comment]
figuring out how to comment on a file that is "in flux" is a fair challenge - tacking a comment to line 12 when someone else is adding many lines of new content at line 10 should not keep the comment tacked to the now completely unrelated new line 12

**Comment by flukeout** [comment]
Yeah, good point - if we do comments we have to figure out what to anchor them to. I think gDocs anchors to a selection.

**Comment by Pomax** [comment]
I guess it also depends on the level of event information we get - if someone adds a comment to line 12, and someone else pastes 7 lines on line 10, then we might be able to see that "this person added a comment, someone else introduce new lines, this person's comment needs to be moved".
